src/components/MonthCalendar.tsx

```diff
@@ -64,40 +64,45 @@ const MonthCalendar = ({ events, onFocusDate, focusDate }: MonthCalendarProps):
             return 0;
         };
 
-        const calendarEvents = getEventsInRange(firstCalendarDay, lastCalendarDay).filter(e => e.campaign.id === campId);
+        const campaignEvents = events.filter(e => e.campaign.id === campId)
+            .sort((a, b) => {
+                return new Date(a.start_time).getTime() - new Date(b.start_time).getTime();
+            });
 
-        if (calendarEvents.length === 0) return { height: 0, top: 0 };
+        if (campaignEvents.length === 0) return { height: 0, top: 0 };
 
-        const firstEventDate = new Date(calendarEvents[0].start_time);
-        const lastEventDate = new Date(calendarEvents[calendarEvents.length - 1].end_time);
+        const firstEventDate = new Date(campaignEvents[0].start_time);
+        const lastEventDate = new Date(campaignEvents[campaignEvents.length - 1].end_time);
 
-        const top = getGridNumber(new Date(firstEventDate)) * barUnit;
-        let height = (getGridNumber(new Date(lastEventDate)) * barUnit) - top;
-        if (height === 0) height = barUnit;
+        let bottom, top;
+        if (firstEventDate < firstCalendarDay ) {
+            top = 0;
+        }
+        else if (firstEventDate > lastCalendarDay) {
+            top = 100;
```

Comment:
What are your thoughts on this @nehahirve?

Reply:
I did make it hidden now if height === 0 > my thoughts were that the 'getcampcaolours' function would get too confusing if it returned either a top and bottom or a display, so i checked for height in the CSS in JS and if height was 0, display is none, is there a better solution?

Reply:
I understand! I agree that it would not be ideal to do very differn

Well, I would probably move all the logic inside `getBarPos()` (and the actual JSX output for the bar) to a separate component or simple render function (within this same module, so no extra file), and return `null` for these conditions.

That way the execution won't go on unnecessarily, and it's also more clear to the reader what happens.